tests/test_run.py

```diff
@@ -48,6 +61,274 @@ async def test_run(async_flag, record_mode):
     assert output[0].url.startswith("https://")
 
 
+@pytest.mark.asyncio
+async def test_run_with_iterator(mock_replicate_api_token):
+    router = respx.Router(base_url="https://api.replicate.com/v1")
+    router.route(method="POST", path="/predictions").mock(
+        return_value=httpx.Response(
+            201,
+            json=_prediction_with_status("starting"),
+        )
+    )
+    router.route(method="GET", path="/predictions/p1").mock(
+        side_effect=[
+            httpx.Response(
+                200,
+                json=_prediction_with_status(
+                    "processing",
+                    [
+                        "Hello, ",
+                    ],
+                ),
+            ),
+            httpx.Response(
+                200,
+                json=_prediction_with_status(
+                    "succeeded",
+                    [
+                        "Hello, ",
+                        "world!",
+                    ],
+                ),
+            ),
+        ]
+    )
+    router.route(
+        method="GET",
+        path="/models/test/example/versions/v1",
+    ).mock(
+        return_value=httpx.Response(
+            201,
+            json=_version_with_schema(
+                "p1",
+                {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                    },
+                    "x-cog-array-type": "iterator",
+                },
+            ),
+        )
+    )
+
+    client = Client(
+        api_token="test-token", transport=httpx.MockTransport(router.handler)
+    )
+    client.poll_interval = 0.001
+
+    stream = cast(
+        Iterator[FileOutput],
+        client.run(
+            "test/example:v1",
+            input={
+                "text": "Hello, world!",
+            },
+            wait=False,
+        ),
+    )
+
```

Comment:
```suggestion
expected = ["Hello, ", "world!"]

for token in stream:
    assert token == expected.pop(0)
assert not expected
```

I don't know... maybe what we have is fine for now.